NumLib/Fem/ShapeFunction/ShapeHex20-impl.h

```diff
@@ -0,0 +1,129 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+namespace
+{
+
+inline double ShapeFunctionHexHQ_Corner(const double r, const double s, const double t)
+{
+    return 0.125 * (1 + r) * (1 + s) * (1 + t) * ( r + s + t - 2.0);
+}
+
+inline double ShapeFunctionHexHQ_Middle(const double r, const double s, const double t)
+{
+    return 0.25 * (1 - r * r) * (1 + s) * (1 + t);
+}
+
+inline double dShapeFunctionHexHQ_Corner(const double r, const double s, const double t, const int ty)
```

Comment:
It is a dimension related indicator, and it may be renamed as ``dim_id`` with a comment "Dimension associated indicator ".